xbmc/utils/Mime.cpp

```diff
@@ -18,501 +18,511 @@
-     {"vmd", "application/vocaltec-media-desc"},
-     {"vmf", "application/vocaltec-media-file"},
-     {"voc", "audio/voc"},
-     {"vos", "video/vosaic"},
-     {"vox", "audio/voxware"},
-     {"vqe", "audio/x-twinvq-plugin"},
-     {"vqf", "audio/x-twinvq"},
-     {"vql", "audio/x-twinvq-plugin"},
-     {"vrml", "application/x-vrml"},
-     {"vrt", "x-world/x-vrt"},
-     {"vsd", "application/x-visio"},
-     {"vst", "application/x-visio"},
-     {"vsw", "application/x-visio"},
-     {"vtt", "text/vtt"},
-     {"w60", "application/wordperfect6.0"},
-     {"w61", "application/wordperfect6.1"},
-     {"w6w", "application/msword"},
-     {"wav", "audio/wav"},
-     {"wb1", "application/x-qpro"},
-     {"wbmp", "image/vnd.wap.wbmp"},
-     {"web", "application/vnd.xara"},
-     {"webp", "image/webp"},
-     {"wiz", "application/msword"},
-     {"wk1", "application/x-123"},
-     {"wma", "audio/x-ms-wma"},
-     {"wmf", "windows/metafile"},
-     {"wml", "text/vnd.wap.wml"},
-     {"wmlc", "application/vnd.wap.wmlc"},
-     {"wmls", "text/vnd.wap.wmlscript"},
-     {"wmlsc", "application/vnd.wap.wmlscriptc"},
-     {"wmv", "video/x-ms-wmv"},
-     {"word", "application/msword"},
-     {"wp", "application/wordperfect"},
-     {"wp5", "application/wordperfect"},
-     {"wp6", "application/wordperfect"},
-     {"wpd", "application/wordperfect"},
-     {"wq1", "application/x-lotus"},
-     {"wri", "application/mswrite"},
-     {"wrl", "model/vrml"},
-     {"wrz", "model/vrml"},
-     {"wsc", "text/scriplet"},
-     {"wsrc", "application/x-wais-source"},
-     {"wtk", "application/x-wintalk"},
-     {"x3f", "image/x3f"},
-     {"xbm", "image/xbm"},
-     {"xdr", "video/x-amt-demorun"},
-     {"xgz", "xgl/drawing"},
-     {"xif", "image/vnd.xiff"},
-     {"xl", "application/excel"},
-     {"xla", "application/excel"},
-     {"xlb", "application/excel"},
-     {"xlc", "application/excel"},
-     {"xld", "application/excel"},
-     {"xlk", "application/excel"},
-     {"xll", "application/excel"},
-     {"xlm", "application/excel"},
-     {"xls", "application/excel"},
-     {"xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"},
-     {"xlt", "application/excel"},
-     {"xlv", "application/excel"},
-     {"xlw", "application/excel"},
-     {"xm", "audio/xm"},
-     {"xml", "text/xml"},
-     {"xmz", "xgl/movie"},
-     {"xpix", "application/x-vnd.ls-xpix"},
-     {"xpm", "image/xpm"},
-     {"x-png", "image/png"},
-     {"xspf", "application/xspf+xml"},
-     {"xsr", "video/x-amt-showrun"},
-     {"xvid", "video/x-msvideo"},
-     {"xwd", "image/x-xwd"},
-     {"xyz", "chemical/x-pdb"},
-     {"z", "application/x-compressed"},
-     {"zip", "application/zip"},
-     {"zoo", "application/octet-stream"},
-     {"zsh", "text/x-script.zsh"}}};
+namespace
+{
+
+const std::map<std::string, std::string>& GetMap()
+{
+  static const std::map<std::string, std::string> mimetypes = {
+      {{"3dm", "x-world/x-3dmf"},
```

Comment:
While on it I suggest to switch to `std::unordered_map` and to remove the superfluous `{}` brackets in line 27 and 520.

Reply:
done.